src/Model/EditorGroup.re

```diff
@@ -82,37 +89,47 @@ let _getAdjacentEditor = (editor: int, reverseTabOrder: list(int)) => {
   };
 };
 
-let setActiveEditorByIndexDiff = (diff, model) => {
-  let tabs = model.reverseTabOrder;
-  let count = List.length(tabs);
-
-  if (count <= 1) {
-    // Nothing to change
-    model
-  } else {
+let setActiveEditorTo = (kind, model) =>
+  switch (model.reverseTabOrder) {
+  | []
+  | [_] => model
+  | _ =>
     switch (model.activeEditorId) {
     | Some(activeEditorId) =>
-      switch (_getIndexOfElement(activeEditorId, tabs)) {
-      | (-1) => model
-      | idx =>
+      switch (_getIndexOfElement(activeEditorId, model.reverseTabOrder)) {
+      | None => model
+      | Some(idx) =>
+        // The diff amounts are inverted because the list is in reverse order
         let newIndex =
-          switch (idx + diff) {
-          // Wrapping negative, go to end
-          | -1 => count - 1
-          // If this is past the end, go to zero, otherwise this index is fine
-          | i => i >= count ? 0 : i
+          switch (kind) {
+          | `Next => idx - 1
+          | `Previous => idx + 1
           };
 
-        {...model, activeEditorId: List.nth_opt(tabs, newIndex)};
+        let count = List.length(model.reverseTabOrder);
+
+        let newIndex =
+          if (newIndex < 0) {
+            // Wrapping negative, go to end
+            count - 1;
+          } else if (newIndex >= count) {
+            0;
+            // If this is past the end, go to zero
+          } else {
+            newIndex;
+          };
```

Comment:
There are some convenience functions in `Core.Utility.IndexEx` for this:

https://github.com/onivim/oni2/blob/7493356cc4bae87343f291c792c4c0fc088fd8b8/src/Core/Utility/IndexEx.re#L1-L27

Reply:
Yep, that is exactly what I need. I'll give it a try when I get a chance.